plugins/@grouparoo/hubspot/src/lib/export-objects/exportRecords.ts

```diff
@@ -221,10 +212,18 @@ export const exportRecords: ExportRecordsPluginMethod = async ({
   exports: recordsToExport,
 }) => {
   const batchExports = buildBatchExports(recordsToExport);
-  return exportBatch({
-    appOptions,
-    destinationOptions,
-    syncOperations,
-    exports: batchExports,
-  });
+  try {
+    return await exportBatch({
+      appOptions,
+      destinationOptions,
+      syncOperations,
+      exports: batchExports,
+    });
+  } catch (error) {
+    if (error?.response?.status === 429) {
```

Comment:
Hmm, this might happen from any of the APIs including search, etc.
Maybe this should be in the client?
Ideally, there might also be headers to read to know when to retry.

Reply:
I got the header.